### PR TITLE
Address bad JNDI configuration described in https://github.com/pegasy…

### PR DESCRIPTION
**Pull request by pegadave**
…stems/docker-pega-web-ready/issues/54

This PR should address the prconfig setting / JNDI conflict described in https://github.com/pegasystems/docker-pega-web-ready/issues/54

### CODE REVIEW

File: tomcat-bin/setenv.sh [hunk 1]
```diff
@@ -50,5 +50,12 @@ CATALINA_OPTS="${CATALINA_OPTS} -DSMAAutoNodeDiscovery=true "
 CATALINA_OPTS="${CATALINA_OPTS} -DSMAAutoNodeDiscoveryJMXPort=${JMX_PORT} "
 CATALINA_OPTS="${CATALINA_OPTS} -DSMAAutoNodeDiscoveryPort=8080 "
 
+# Provide setting required for stream node 
+if [ "${NODE_TYPE}" = "Stream" ]; then
```
Comment (dcasavant) [inline tomcat-bin/setenv.sh]: The node_type parameter could contain several different node classifications (such as "WebUser,Backgroundprocessing,Stream".  Can we use the env var IS_STREAM_NODE?

**Comment by Emilio-Pega** [comment]
Out of curiosity, but wouldn't the same fundamental issue occur for any of the prconfig settings defined here, not just the stream ones?

**Comment by Emilio-Pega** [comment]
For example, why are these immune to the underlying issue (and not changed in this pr)?

```
 <Environment name="prconfig/session/ha/enabled" value="true" type="java.lang.String"/>	  <Environment name="prconfig/session/ha/enabled" value="true" type="java.lang.String"/>
  <Environment name="prconfig/dsm/services/stream/pyBaseLogPath" value="/opt/pega/kafkadata" type="java.lang.String" />
```

**Comment by dcasavant** [comment]
@Emilio-Pega The problem seems to stem from settings that have a name that is the root of another name.  For example, if this would only be a problem if there were settings `a/b` and `a/b/c`.

**Comment by Emilio-Pega** [comment]
OK, that makes sense.

**Comment by dcasavant** [comment]
This fixes #54.

**Comment by pegadave** [comment]
@Emilio-Pega it is only an issue if you have JNDI names like: 

abc/feature (some value) (bound to an object that is NOT a javax.naming.Context)
abc/feature/size (some value)

Were we to update our application to look for the following names instead:

abc/feature/enabled (some value)
abc/feature/size (some value)

Things should be fine.

General guide lines for JNDI names being used for this purpose (prconfig settings) would be:
1. JNDI names should be unique.
2. JNDI names bound to an object can't correspond exactly to a subcontext of another object in the JNDI namespace.

To be clear, 'object' in #2 above assumes that you're talking about something that is not an implementation of javax.naming.Context.

**Comment by Emilio-Pega** [comment]
yup makes total sense.  should we put a comment in the file to that effect for us (the devs) so we don't forget and make the same mistake twice?

**Comment by pegadave** [comment]
@Emilio-Pega We will have to be vigilant about the JNDI config we specify in context.xml and prweb.xml as there are likely many potential case for this to be an issue.  That said there are any others in the Pega image at this time.

**Comment by pegadave** [comment]
@Emilio-Pega -- that's a good idea. I will add comments to that effect in context.xml and prweb.xml.

**Pull request merged**